submissions/5mountains/tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,115 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Resident {
+  constructor(species, name, gender) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+  }
+  toString() {
+    return `Wonderful creature - <b>${
+      this.species
+    }</b>, whose name is <b>${
+      this.name
+    }</b> gender is <b>${
+      this.gender
+    }</b>!`;
+  }
+}
+
+class Human extends Resident {
+  constructor(name, gender, saying, friends = 'no one') {
+    super('human', name, gender);
+    this.legs = 2;
+    this.hands = 2;
+    this.saying = saying;
+    this.friends = friends;
+  }
+  toString() {
+    return (
+      super.toString() + `</b> Usually the greeting is <b>${
+        this.saying
+      }</b> The friends of this inhabitant are <b>${
+        this.friends.join(', ')
+      }</b>. By the wat, this member is owner of <b>${
+        this.legs
+      } legs</b> and <b>${
+        this.hands
+      } hands</b>!`
+    );
+  }
+}
+
+class Animal extends Resident {
+  constructor(species, name, gender, saying, friends = 'no one') {
+    super(species, name, gender);
+    this.paws = 4;
+    this.saying = saying;
+    this.friends = friends;
+  }
+  toString() {
+    return super.toString() + `</b> Usually the greeting is <b>${
+      this.saying
+    }</b> The friends of this inhabitant are <b>${
+      this.friends.join(', ')
+    }</b>. By the wat, this member is owner of <b>${
+     this.paws
+    } paws</b>!`;
+  }
+}
+
+class Dog extends Animal {
+  constructor(name, gender, saying, friends) {
+    super('dog', name, gender, saying, friends);
+    this.saying =  'woof-woof!';
+    this.friends = ['Tom', 'Jenny'];
+  }
+}
+
+class Cat extends Animal {
+  constructor(name, gender, saying, friends) {
+    super('cat', name, gender, saying, friends);
+    this.saying = 'meow!';
+    this.friends =  ['Tom', 'Jenny', 'Selina'];

```

Comment:
This says every cat will have Tom, Jenny and Selina as their friends. Not realistic scenario.

Reply:
I think I fixed it :)